cpp/tests/collect_list/collect_list_test.cpp

```diff
@@ -466,6 +636,87 @@ TYPED_TEST(TypedCollectListTest, BasicGroupedTimeRangeRollingWindow)
     {21, 22, 23}}.release();
 
   CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(expected_result->view(), result->view());
+
+  auto const result_with_nulls_excluded =
+    grouped_time_range_rolling_window(table_view{std::vector<column_view>{group_column}},
+                                      time_column,
+                                      cudf::order::ASCENDING,
+                                      input_column,
+                                      preceding,
+                                      following,
+                                      min_periods,
+                                      make_collect_aggregation(null_policy::EXCLUDE));
+
+  CUDF_TEST_EXPECT_COLUMNS_EQUIVALENT(expected_result->view(), result_with_nulls_excluded->view());
+}
+
+TYPED_TEST(TypedCollectListTest, GroupedTimeRangeRollingWindowWithNulls)
+{
+  using namespace cudf;
+  using namespace cudf::test;
+
+  using T = TypeParam;
+
+  auto const time_column = fixed_width_column_wrapper<cudf::timestamp_D, cudf::timestamp_D::rep>{
+    1, 1, 2, 2, 3, 1, 4, 5, 6};
+  auto const group_column = fixed_width_column_wrapper<int32_t>{1, 1, 1, 1, 1, 2, 2, 2, 2};
+  auto const input_column = fixed_width_column_wrapper<T, int32_t>{
+    {10, 11, 12, 13, 14, 20, 21, 22, 23}, {1, 0, 1, 1, 1, 1, 0, 1, 1}};
+  auto const preceding   = 2;
+  auto const following   = 1;
+  auto const min_periods = 1;
+  auto const result =
+    grouped_time_range_rolling_window(table_view{std::vector<column_view>{group_column}},
+                                      time_column,
+                                      cudf::order::ASCENDING,
+                                      input_column,
+                                      preceding,
+                                      following,
+                                      min_periods,
+                                      make_collect_aggregation());
+
+  auto null_at = [](auto null_idx) {
+    return make_counting_transform_iterator(0, [null_idx](auto i) { return i != null_idx; });
+  };
```

Comment:
This is neat. IMO it would be nice to have it as a utility function (`iterator_with_null_at` or something like that).
A version that takes a vector of values (not that fast, but it's meant for small input AFAICT) could also be added.

Reply:
> A version that takes a vector of values...

Thanks for confirming. I thought that would be useful too. I'll see about adding one to `cudf::test`.

Reply:
Sorry for the delay. I've added this utility function.